samples/batch/transcription-enabled-storage/Connector/StorageUtilities.cs

```diff
@@ -0,0 +1,187 @@
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+// </copyright>
+
+namespace Connector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.WindowsAzure.Storage;
+    using Microsoft.WindowsAzure.Storage.Blob;
+
+    public static class StorageUtilities
+    {
+        public static async Task<string> CreateSASAsync(string storageConnectionString, string filePath, ILogger log)
+        {
+            var containerName = GetContainerNameFromPath(filePath);
+            var storageAccount = CloudStorageAccount.Parse(storageConnectionString);
+            var blobClient = storageAccount.CreateCloudBlobClient();
+            var container = blobClient.GetContainerReference(containerName);
+
+            var containerPermissions = new BlobContainerPermissions
+            {
+                PublicAccess = BlobContainerPublicAccessType.Blob
+            };
+
+            await container.SetPermissionsAsync(containerPermissions).ConfigureAwait(false);
+            containerPermissions.SharedAccessPolicies.Add("mypolicy", new SharedAccessBlobPolicy()
+            {
+                SharedAccessStartTime = DateTime.UtcNow,
+                SharedAccessExpiryTime = DateTime.UtcNow.AddDays(1),
+                Permissions = SharedAccessBlobPermissions.Read
+            });
+
+            var sas = container.GetSharedAccessSignature(containerPermissions.SharedAccessPolicies.FirstOrDefault().Value);
+            var absoluteSasUri = filePath + sas;
+
+            log.LogInformation($"Created sas url for file {filePath}");
+            return absoluteSasUri;
+        }
+
+        public static async Task WriteTextFileToBlobAsync(string storageConnectionString, string content, string containerName, string fileName, ILogger log)
+        {
+            log.LogInformation($"Writing file {fileName} to container {containerName}.");
+
+            var storageAccount = CloudStorageAccount.Parse(storageConnectionString);
+            var fileClient = storageAccount.CreateCloudBlobClient();
+            var container = fileClient.GetContainerReference(containerName);
+            var blockBlob = container.GetBlockBlobReference(fileName);
+
+            await blockBlob.UploadTextAsync(content).ConfigureAwait(false);
+        }
+
+        public static AudioDetails GetAudioDetails(byte[] audioBytes, string fileName, string locale, DateTime transcriptionCreationTime, bool isCustomModel, bool sentimentAnalysisAdded, bool entityRedactionAdded, ILogger log)
+        {
+            var fileNameExtension = Path.GetExtension(fileName);
+
+            var duration = TimeSpan.Zero;
+            var channels = 0;
+
+            try
+            {
+                duration = AudioFileProcessor.GetDuration(audioBytes, fileNameExtension);
```

Comment:
The duration in the wav header might not match the real audio length. We need to be careful here to avoid return different values than what speech transcription returns.

Reply:
Fully agree! That's why I removed the receipt file (and therefore getting the audio length from file header as well)